### PR TITLE
Store is now created by ngRedux (allows DI in middleware)

### PR DESCRIPTION
**Pull request by wbuchwalter**

### CODE REVIEW

File: src/components/ngRedux.js [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+  let _middlewares = [];
+  let _storeEnhancer = undefined;
 
-  this.$get = () => {
-    return Connector(reduxStore);
+  this.createStoreWith = (reducer, middlewares, storeEnhancer) => {
+  	  invariant(
+        isFunction(reducer),
+        'The reducer parameter passed to createStoreWith must be a Function. Instead received %s.',
+        typeof selector
```
Comment (e-schultz) [inline src/components/ngRedux.js]: should this be:

```javascript
	  invariant(
        isFunction(reducer),
        'The reducer parameter passed to createStoreWith must be a Function. Instead received %s.',
        typeof reducer
      );
```

noticed a similar error just below here, and in connector also. Ends up having the error be `received undefined`  even if passing in object/string/etc.

Comment (wbuchwalter) [inline src/components/ngRedux.js]: absolutely...

**Comment by wbuchwalter** [comment]
Fixed

File: src/components/ngRedux.js [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
+
+      _reducer = reducer;
+      _storeEnhancer = storeEnhancer || createStore;
+      _middlewares = middlewares;
+  };
+
+  this.$get = ($injector) => {
+  	let resolvedMiddleware = [];
+  	for(let middleware of _middlewares) {
+  		if(_.isString(middleware)) {
```
Comment (e-schultz) [inline src/components/ngRedux.js]: `_` is undefined 

but when changing it to `typeof middleware === string`, I was able to get it working.

Made a simple log middleware that used angular $log instead, and was able to get things working.

```javascript
.factory('logger', ['$log', function ($log) {
    return store => next => action => {
      $log.log('dispatching', action);
      let result = next(action);
      $log.log('next state', store.getState());
      return result;
    };

  }])
  .config(['$ngReduxProvider', ($ngReduxProvider) => {
    //let createStoreWithMiddleware = applyMiddleware(apiMiddleware, logger, thunk)(createStore);
    //let store = createStoreWithMiddleware(reducers);

    //$ngReduxProvider.setReduxStore(store);
    $ngReduxProvider.createStoreWith(reducers, [apiMiddleware, 'logger', thunk]);
  }])
```

Comment (wbuchwalter) [inline src/components/ngRedux.js]: Hmm that's weird, I added lodash to the package.json in this PR, maybe you forgot to redo npm install?

Comment (e-schultz) [inline src/components/ngRedux.js]: missing an `import 'lodash';`, added that and it worked.

Comment (wbuchwalter) [inline src/components/ngRedux.js]: Allright, sorry for all this errors, I didn't had time to run my code yesterday.
I removed lodash, it was a bit overkill for a typeof.

I will merge and publish by midday (EST).
Thanks!

Comment (e-schultz) [inline src/components/ngRedux.js]: No problem, happy to help.

Thanks for making this change, redux + ng-redux is looking really promising for a project that I'm just starting.

**Pull request merged**